src/main/java/com/commercetools/sync/shoppinglists/utils/LineItemUpdateActionUtils.java

```diff
@@ -0,0 +1,257 @@
+                // To be able to ensure the order, we need to remove and add this line item back
+                // with the up to date values.
+                indexOfFirstDifference = i;
+                break;
+            }
+        }
+
+        // for example:
+        // old: li-1, li-2
+        // new: li-1, li-3, li-2
+        // indexOfFirstDifference: 1 (li-2 vs li-3)
+        // expected: remove from old li-2, add from draft li-3, li-2 starting from the index.
+        for (int i = indexOfFirstDifference; i < oldLineItems.size(); i++) {
+            updateActions.add(RemoveLineItem.of(oldLineItems.get(i).getId()));
+        }
+
+        for (int i = indexOfFirstDifference; i < newlineItems.size(); i++) {
+            updateActions.add(AddLineItemWithSku.of(newlineItems.get(i)));
+        }
+
+        return updateActions;
+    }
+
+    private static boolean hasIdenticalAddedAtValues(
+        @Nonnull final LineItem oldLineItem,
+        @Nonnull final LineItemDraft newLineItem) {
+
+        if (newLineItem.getAddedAt() == null) {
+            return true; // omit, if not set in draft.
+        }
+
+        return oldLineItem.getAddedAt().equals(newLineItem.getAddedAt());
+    }
+
+    /**
+     * Compares all the fields of a {@link LineItem} and a {@link LineItemDraft} and returns a list of
+     * {@link UpdateAction}&lt;{@link ShoppingList}&gt; as a result. If both the {@link LineItem} and
+     * the {@link LineItemDraft} have identical fields, then no update action is needed and hence an empty {@link List}
+     * is returned.
+     *
+     * @param oldShoppingList shopping list resource, whose line item should be updated.
+     * @param newShoppingList new shopping list draft, which contains the line item to update.
+     * @param oldLineItem     the line item which should be updated.
+     * @param newLineItem     the line item draft where we get the new fields (i.e. quantity, custom fields and types).
+     * @param syncOptions     responsible for supplying the sync options to the sync utility method. It is used for
+     *                        triggering the error callback within the utility, in case of errors.
+     * @return A list with the update actions or an empty list if the line item fields are identical.
+     */
+    @Nonnull
+    public static List<UpdateAction<ShoppingList>> buildLineItemUpdateActions(
+        @Nonnull final ShoppingList oldShoppingList,
+        @Nonnull final ShoppingListDraft newShoppingList,
+        @Nonnull final LineItem oldLineItem,
+        @Nonnull final LineItemDraft newLineItem,
+        @Nonnull final ShoppingListSyncOptions syncOptions) {
+
+        final List<UpdateAction<ShoppingList>> updateActions = filterEmptyOptionals(
+            buildChangeLineItemQuantityUpdateAction(oldLineItem, newLineItem)
+        );
+
+        updateActions.addAll(
+            buildLineItemCustomUpdateActions(oldShoppingList, newShoppingList, oldLineItem, newLineItem, syncOptions));
+
```

Comment:
Do we need not to cater **SetLineItemCustomField**?

Reply:
It's part of it.